contrib/database/sql/sql.go

```diff
@@ -12,49 +12,169 @@
 package sql
 
 import (
+	"context"
 	"database/sql"
 	"database/sql/driver"
 	"errors"
+	"math"
+	"reflect"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql/internal"
 )
 
+// registeredDrivers holds a registry of all drivers registered via the sqltrace package.
+var registeredDrivers = &driverRegistry{
+	keys:    make(map[reflect.Type]string),
+	drivers: make(map[string]driver.Driver),
+	configs: make(map[string]*config),
+}
+
+type driverRegistry struct {
+	// keys maps driver types to their registered names.
+	keys map[reflect.Type]string
+	// drivers maps keys to their registered driver.
+	drivers map[string]driver.Driver
+	// configs maps keys to their registered configuration.
+	configs map[string]*config
+}
+
+// registered checks if the named driver has already been registered.
+func (d *driverRegistry) registered(name string) bool {
+	_, ok := d.configs[name]
+	return ok
+}
+
+// add adds the driver and config to the driver registy.
```

Comment:
```suggestion
// add adds the driver and config to the driver registry.
```

Reply:
Should we rephrase this to clarify that the config and name belong to the driver? The way it's phrased now it could mean that config is something unrelated. Maybe something like:
```go
// add adds the driver with the given name and config to the registry.
```